lib/ssz_ex.ex

```diff
@@ -80,78 +80,97 @@ defmodule LambdaEthereumConsensus.SszEx do
       else: decode_fixed_container(binary, module)
   end
 
-  @spec hash_tree_root!(boolean, atom) :: Types.root()
-  def hash_tree_root!(value, :bool), do: pack(value, :bool)
+  @spec hash_tree_root!(any, any) :: Types.root()
+  def hash_tree_root!(value, schema) do
+    {:ok, root} = hash_tree_root(value, schema)
+    root
+  end
 
-  @spec hash_tree_root!(non_neg_integer, {:int, non_neg_integer}) :: Types.root()
-  def hash_tree_root!(value, {:int, size}), do: pack(value, {:int, size})
+  @spec hash_tree_root(boolean, atom) :: Types.root()
+  def hash_tree_root(value, :bool), do: {:ok, pack(value, :bool)}
```

Comment:
Don't we need to merkleize after packing?

Reply:
For ints and bools, we don't because it's a single chunk

Reply:
But the spec says:

> * `merkleize(pack(value))` if `value` is a basic object or a vector of basic objects.

Here we're only doing `pack(value)` 🤔

Reply:
Here:

> Then, merkleize the chunks (empty input is padded to 1 zero chunk):
- If 1 chunk: the root is the chunk itself.
- If > 1 chunks: merkleize as binary tree.

Reply:
So `hash_tree_root(3) == <<3::little-size(256>>` ?

Reply:
Not exactly, but yes

Reply:
🤯